asteroid/data/avspeech_dataset.py

```diff
@@ -0,0 +1,238 @@
+        self.audio_path = Path(audio_path)
+        self.video_start_length = video_start_length
+
+        self.embed_path = None
+        self.embed_saved = False
+        self.embed = None
+        self.embed_dir = embed_dir
+
+        self.load_spec = load_spec
+        self._is_spec = False
+
+        self.fps = fps
+        self.signal_len = signal_len
+        self.total_frames = self.fps * self.signal_len
+        
+        self.sr = sr
+        self.total_audio_frame = self.sr * self.signal_len
+
+        self.spec_path = Path(*self.audio_path.parts[:-2], "spec",
+                              self.audio_path.stem + ".npy")
+
+        self._load(sr=sr)
+        self._check_video_embed()
+        self._convert_video()
+
+    def _load(self, sr: int):
+        self.audio = None
+        if self.load_spec and self.spec_path.is_file():
+            self.audio = np.load(self.spec_path)
+            self._is_spec = True
+        if self.audio is None:
+            self.audio, sr = librosa.load(self.audio_path.as_posix(), sr=sr)
+            self._is_spec = False
+        self.video = cv2.VideoCapture(self.video_path.as_posix())
+
+    def _convert_video(self):
+        if self.embed_saved:
+            return
+        self.buffer_video = get_frames(self.video)
+
+    def _check_video_embed(self, embed_ext=".npy"):
+        video_name_stem = self.video_path.stem
+
+        embed_dir = self.embed_dir
+        if not embed_dir.is_dir():
+            # check embed_dir="../../dir" or embed_dir="dir"
+            embed_dir = Path(*embed_dir.parts[2:])
+
+        self.embed_path = Path(embed_dir, f"{video_name_stem}_part{self.video_start_length}{embed_ext}")
+        if self.embed_path.is_file():
+            self.embed_saved = True
+            self.embed = np.load(self.embed_path.as_posix())
+
+    def embed_is_saved(self):
+        return self.embed_saved
+
+    def get_embed(self):
+        return self.embed
+
+    def get_video(self):
+        #retrieve slice of video, if video > 75 frames
+        buffer_video = self.buffer_video[self.video_start_length*self.total_frames:\
+                                        (self.video_start_length+1)*self.total_frames]
+        return buffer_video
+
+    def get_audio(self):
+        return self.audio
+
+    def get_spec(self):
+            return np.load(self.spec_path)
+
+    def is_spec(self):
+        return self._is_spec
+
+    @staticmethod
+    def load_audio(audio_path: str, sr=16_000, load_spec=False):
+        audio_path = Path(audio_path)
+        spec_exists = False
+        spec_path = Path(*audio_path.parts[:-2], "spec", audio_path.stem + ".npy")
+        if load_spec and spec_path.is_file():
+            spec_exists = True
+            audio = np.load(spec_path)
+        else:
+            audio = librosa.load(str(audio_path), sr=sr)[0]
+        return audio, spec_exists, spec_path
+
+
+class AVSpeechDataset(data.Dataset):
+
+    def __init__(self, input_df_path: Path,
```

Comment:
Input arguments should also accept strings, it's a simple check in the input (if str, convert to Path).